pkg/kubectl/cmd/util/factory.go

```diff
@@ -307,11 +308,28 @@ type schemaClient interface {
 	Get() *client.Request
 }
 
+func substituteUser(dir string) (string, error) {
```

Comment:
IMO, we should do the same thing that we do for kubeconfig, which currently is to just use [os.Getenv(HOME)](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/clientcmd/loader.go#L42). If the goal is to be platform agnostic (*cough*windows*cough*), then I'd recommend we put this logic in the loader and sharing it for both schema caching and kubeconfig.

Also, it wasn't clear to me from docs whether `user.Current()` will detect an overwritten `$HOME`. Probably safest to do `os.GetEnv(HOME)` and fallback to this if it isn't set, so we don't break existing behavior.